RGPosadas/LitterRally
# TASK-29: Login Backend

This PR closes #111.

**Steps to Test Local**

1. Start the server using ```nx serve mull-api```
2. Open the GraphQL playground.
3. Click the Settings cog on the top right
4. Change the ```request.credentials``` setting to "include"
 Like so: 
![image](https://user-images.githubusercontent.com/36024638/102433269-8cf52680-3fe1-11eb-80f1-6c12581d8e37.png)
5. Open your browser dev tools in the ```Application``` tab
6. Under Storage, click on cookies for the server domain (http://localhost:3333)
7. Create a user using a mutation or mull-ui
8. Login using the ```login``` mutation
9. An accessToken should be returned and a cookie named ```mullToken``` should be set.

**Steps to Test OAuth**
1. Start the server using ```nx serve mull-api```
2. Open your browser and go to ```http://localhost:3333/api/auth/google | twitter | facebook/```
3. An accessToken should be returned and a cookie named ```mullToken``` should be set.

**Notes**
Refer to the sequence diagrams on [lucidchart](https://lucid.app/lucidchart/invitations/accept/6afcb749-5988-4bdb-9def-865d6dcf9ad9) for more information.
This requires an update to the .env file. Updated .env available on [Slack](https://bug-bytes.slack.com/archives/CS5P4NL5Q/p1606083803001700).
